v2/thirdpartypasswordless/common-customizations/change-email.mdx

```diff
@@ -0,0 +1,983 @@
+    );
+    return regexp.test(email);
+}
+
+```
+
+
+</TabItem>
+
+<TabItem value="go">
+
+```go
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"regexp"
+
+	"github.com/supertokens/supertokens-golang/ingredients/emaildelivery"
+	"github.com/supertokens/supertokens-golang/recipe/emailverification"
+	"github.com/supertokens/supertokens-golang/recipe/session"
+	"github.com/supertokens/supertokens-golang/recipe/thirdpartypasswordless"
+)
+
+type RequestBody struct {
+	email string
+}
+
+// the following example uses net/http
+func main() {
+	_ = http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		// Wrap the API handler in session.VerifySession
+		session.VerifySession(nil, changeEmailAPI).ServeHTTP(rw, r)
+	})
+}
+
+func changeEmailAPI(w http.ResponseWriter, r *http.Request) {
+	// highlight-start
+	sessionContainer := session.GetSessionFromRequestContext(r.Context())
+
+	var requestBody RequestBody
+	err := json.NewDecoder(r.Body).Decode(&requestBody)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	// validate the input email
+	if !isValidEmail(requestBody.email) {
+		// TODO: handle invalid email error
+		return
+	}
+
+	// check that the account is not a social account
+	userId := sessionContainer.GetUserID()
+	userAccount, err := thirdpartypasswordless.GetUserById(userId)
+	if err != nil {
+		// TODO: handle error
+		return
+	}
+
+	if userAccount.ThirdParty != nil {
+		// TODO: handle error, cannot update email for third party accounts
+		return
+	}
+
+	// Check if the new email is already associated with another email-password user.
+	// If it is, then we throw an error. If it's already associated with this user,
+	// then we return a success response with an appropriate message.
+	existingUsers, err := thirdpartypasswordless.GetUsersByEmail(requestBody.email)
+
+	if err != nil {
+		// TODO: handle error
+	}
+
+	if len(existingUsers) > 0 {
+		for i := 0; i < len(existingUsers); i++ {
+			if existingUsers[i].ID == userId {
+				// TODO: send successful response, email already belongs to this account.
+				return
+			}
+		}
+		// TODO: handle error, email already exists with another account
+	}
+
+	// Then, we check if the email is verified for this user ID or not.
+	// It is important to understand that SuperTokens stores email verification
+	// status based on the user ID AND the email, and not just the email.
+	isVerified, err := emailverification.IsEmailVerified(userId, &requestBody.email)
+
+	if err != nil {
+		// TODO: handle error
+	}
+
+	if !isVerified {
+		// Now we create and send the email verification link to the user for the new email.
+		tokenInfo, err := emailverification.CreateEmailVerificationToken(userId, &requestBody.email)
+		if err != nil {
+			// TODO: handle error
+		}
+
+		if tokenInfo.OK != nil {
+			link := "^{form_apiDomain}^{form_apiBasePath}/verify-email?token=" + tokenInfo.OK.Token
```

Comment:
it's not the api domain where the link goes to. It's the website domain..

Please make this change everywhere and check also in the update email guide in emailpassword and thirdpartyemailpassword recipe (since i assume you copied from there)